web/modules/comlink/src/index.ts

```diff
@@ -0,0 +1,124 @@
+import * as Comlink from "comlink"
+/**
+ * Releases a remote proxy. This is required to prevent memory leaks.
+ *
+ * @param remote - The proxy to release.
+ */
+export function releaseRemote<T>(remote: Comlink.Remote<T>) {
+  if (remote[Comlink.releaseProxy]) remote[Comlink.releaseProxy]()
+}
+
+export const transfer = Comlink.transfer
+
+export type {
+  Endpoint,
+  Local,
+  LocalObject,
+  ProxyMarked,
+  ProxyMethods,
+  ProxyOrClone,
+  Remote,
+  RemoteObject,
+  TransferHandler,
+  UnproxyOrClone
+} from "comlink"
+export { Comlink }
+
+export abstract class AbstractWorkerBase<T> {
+  /** True if the worker has already been terminated. */
+  declare terminated: boolean
+
+  /** Tracks if the worker is still be created. Prevents a race condition. */
+  declare starting?: Promise<void>
+
+  /** Required function needed for getting a `Worker` or `Comlink.Remote<T>` instance. */
+  protected abstract createWorker(): Promisable<Worker | Comlink.Remote<T> | false>
+
+  /** The worker instance. */
+  declare worker?: Comlink.Remote<T>
+
+  constructor() {
+    this.terminated = false
+  }
+
+  /**
+   * Function intended to be used within an `extends` expression.
+   * Constructs a class that wraps around a worker factory function and
+   * automatically handles binding methods and worker creation.
+   *
+   * @param methods - The methods to bind to the class instance, which when
+   *   called will be passed the worker instance.
+   */
+  static of<T>(
+    methods: (keyof T)[]
+  ): abstract new () => AbstractWorkerBase<T> & Comlink.RemoteObject<T> {
+    // @ts-ignore
+    return class extends AbstractWorkerBase<T> {
```

Comment:
You can do this? Incredible

Reply:
yeah honestly when I got this horrifying thing working I was very surprised